tfjs-backend-webgpu/src/matmul_packed_webgpu.ts

```diff
@@ -211,26 +209,28 @@ export function makeMatMulPackedVec4Source(
             let inputCol = tileCol;
             mm_Bsub[inputRow][inputCol] = mm_readB(batchB, kStart + inputRow, globalCol);
         }
-        kStart = kStart + tileInner;
+        kStart = kStart + ${tileInner};
         workgroupBarrier();
 
         // Compute acc values for a single thread.
-        for (var k = 0; k < tileInner / innerElementSize; k = k + 1) {
-            let BCached0 = mm_Bsub[k * innerElementSize][tileCol];
-            let BCached1 = mm_Bsub[k * innerElementSize + 1][tileCol];
-            let BCached2 = mm_Bsub[k * innerElementSize + 2][tileCol];
+        for (var k = 0; k < ${tileInner} / ${innerElementSize}; k = k + 1) {
+            let BCached0 = mm_Bsub[k * ${innerElementSize}][tileCol];
+            let BCached1 = mm_Bsub[k * ${innerElementSize} + 1][tileCol];
+            let BCached2 = mm_Bsub[k * ${innerElementSize} + 2][tileCol];
             ${
       innerElementSize === 3 ?
           '' :
-          'let BCached3 = mm_Bsub[k * innerElementSize + 3][tileCol];'}
+          `let BCached3 = mm_Bsub[k * ${innerElementSize} + 3][tileCol];`}
 
-            ${calculateResultSnippet(transposeA, innerElementSize)}
+            ${
+      calculateResultSnippet(transposeA, innerElementSize, rowPerThread)}
         }
 
         workgroupBarrier();
     }
 
-    for (var innerRow = 0; innerRow < rowPerThread; innerRow = innerRow + 1) {
+    for (var innerRow = 0; innerRow < ${
+      rowPerThread}; innerRow = innerRow + 1) {
```

Comment:
clang format really makes code hard to read. Can we disable clang format for shader code?
// clang-format off
// shader code
// clang-format on

@qjia7 @hujiajie @xhcao What do you think?

Reply:
Sounds good for me.

Reply:
I do not quite love exceptions. Usually I try to fix such issues with less notable techniques.

I remember `innerRow++` works in Tint, does it work in naga?

Reply:
Seems `innerRow++` works, but `++innerRow` does not work...
I thought Yang means the weird newline, should we start from this PR, or in another PR for ALL?

Reply:
WGSL only supports i++. Let's keep this PR as is, and have offline discussion about this.

Reply:
Please try to use i++ for weird new lines in this PR, to see if we can correct some of them.

Reply:
Done.